pennylane/measurements/expval.py

```diff
@@ -104,8 +104,8 @@ def process_samples(
         shot_range: Tuple[int] = None,
         bin_size: int = None,
     ):
-        if isinstance(self.obs, Projector):
-            # branch specifically to handle the projector observable
+        if isinstance(self.obs, Projector) and len(self.obs.parameters[0]) == len(self.obs.wires):
```

Comment:
perhaps we should save/expose this information on the observable somehow, rather than computing it every time we need it?

Reply:
We could actually check `isinstance(self.obs, _BasisStateProjector)`. We originally did it this way to avoid some circular dependencies in other parts of the library, but it would be safe here and much more readable too.